bundles/org.eclipse.passage.lic.equinox/src/org/eclipse/passage/lic/internal/equinox/io/BundleKeyKeeper.java

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************************
+ * Copyright (c) 2020 ArSysOp
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ArSysOp - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.lic.internal.equinox.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import org.eclipse.passage.lic.internal.api.LicensedProduct;
+import org.eclipse.passage.lic.internal.api.LicensingException;
+import org.eclipse.passage.lic.internal.api.io.KeyKeeper;
+import org.eclipse.passage.lic.internal.base.io.FileNameFromLicensedProduct;
+import org.eclipse.passage.lic.internal.base.io.PassageFileExtension;
+import org.eclipse.passage.lic.internal.equinox.i18n.EquinoxMessages;
+import org.osgi.framework.Bundle;
+
+/**
+ * Look for the product's public key into OSGI-INF folder of the configured
+ * bundle.
+ */
+@SuppressWarnings("restriction")
+public final class BundleKeyKeeper implements KeyKeeper {
+
+	private final Supplier<LicensedProduct> product;
+	private final Bundle bundle;
+
+	public BundleKeyKeeper(Supplier<LicensedProduct> product, Bundle bundle) {
+		Objects.requireNonNull(product, "BundleKeyKeeper::product"); //$NON-NLS-1$
+		Objects.requireNonNull(bundle, "BundleKeyKeeper::bundle"); //$NON-NLS-1$
+		this.product = product;
+		this.bundle = bundle;
+	}
+
+	@Override
+	public LicensedProduct id() {
+		return product.get();
+	}
+
+	@Override
+	public InputStream productPublicKey() throws LicensingException {
+		URL resource = resource(Paths.get("OSGI-INF").resolve(keyFile())); //$NON-NLS-1$
```

Comment:
Should we have `PassageBundlePath` later to encapsulate a way to find the info like this?
We may want to store the licensing information under a subfolder later, like they store bundle properties under `OSGI-INF/l10n`

Reply:
Thought about that, but lots of questions arised. This implementation sticks to the current functionality to keep status quo where it's possible. Let's do it when we actually will need it.